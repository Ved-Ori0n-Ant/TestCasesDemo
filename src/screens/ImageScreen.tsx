import React from 'react';
import {Image, SafeAreaView, StyleSheet, View} from 'react-native';
import NextnPrevious, {Stack} from '../components/NextnPrevious';

const ImageScreen = () => {
  return (
    <SafeAreaView style={styles.body}>
      <View style={styles.subBody}>
        <Image
          source={{uri: 'https://reactnative.dev/docs/assets/p_cat2.png'}}
          style={{
            height: 200,
            width: 200,
          }}
          testID={'Image'}
        />
      </View>
      <NextnPrevious currentScreen={Stack.ImageScreen} />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  body: {
    flex: 1,
  },
  subBody: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default ImageScreen;
