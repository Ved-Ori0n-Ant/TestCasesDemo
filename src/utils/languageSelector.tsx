import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Dimensions,
  Image,
} from 'react-native';
import {useTranslation} from 'react-i18next';
import '../localization';

const {width} = Dimensions.get('window');

const LANGS = [
  {lngCode: 'en', label: 'English'},
  {lngCode: 'hi', label: 'हिन्दी'},
  {lngCode: 'ar', label: 'عربي'},
];

const LanguageSelector = () => {
  const {t, i18n} = useTranslation();
  const selectedLngCode: any = i18n.language;
  const setLng = (lngCode: string) => i18n.changeLanguage(lngCode);
  return (
    <View style={styles.container}>
      <View style={styles.titleContainer}>
        {/* Should return select your language in respective language */}
        <Text style={styles.select}>{t('languageSelector:selectLng')}</Text> 
      </View>
      {LANGS.map((l) => {
        const selected = l.lngCode === selectedLngCode;
        return (
          <TouchableOpacity
            onPress={() => setLng(l.lngCode)}
            key={l.lngCode}
            disabled={selected}>
            <View style={[styles.row, selected ? styles.selectedRow : {}]}>
              <Text style={[selected ? styles.selectedText : styles.text]}>
                {l.label}
              </Text>
              {selected && <Image source={require('../utils/images/world.png')} style={{height: 20, width:20}} />}
            </View>
          </TouchableOpacity>
        );
      })}
    </View>
  );
};

export default LanguageSelector;

const styles = StyleSheet.create({
  container: {
    padding: 10,
    backgroundColor: 'rgb(231, 232, 235)',
    width: width * 0.8,
  },
  titleContainer: {
    alignItems: 'center',
    paddingBottom: 20,
  },
  select: {
    fontSize: 20,
  },
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 10,
    paddingVertical: 5,
  },
  selectedRow: {
    backgroundColor: 'rgb(45, 45, 45)',
  },
  selectedText: {
    color: 'rgb(231, 232, 235)',
  },
  text: {
    color: 'rgb(45, 45, 45)',
  },
});