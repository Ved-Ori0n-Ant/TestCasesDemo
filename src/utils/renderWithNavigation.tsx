import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {render} from '@testing-library/react-native';

export function renderWithNavigation(renderComponent: any) {
  return render(<NavigationContainer>{renderComponent}</NavigationContainer>);
}
