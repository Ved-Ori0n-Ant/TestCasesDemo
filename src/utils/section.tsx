import React, {ReactNode} from 'react';
import {StyleSheet, View, Text} from 'react-native';

interface Props {
  children: ReactNode;
  title: string;
  testID?: string;
}

const Section = ({title, children, testID}: Props) => {
  return (
    <View style={styles.container} testID={testID}>
      <View style={styles.titleContainer}>
        <Text style={styles.title}>{title}</Text>
      </View>
      {children}
    </View>
  );
};

export default Section;

const styles = StyleSheet.create({
  container: {
    alignContent: 'center',
    marginVertical: 10,
    padding: 10,
    justifyContent: 'center',
    width: '80%',
  },
  titleContainer: {
    alignItems: 'center',
  },
  title: {
    fontSize: 26,
    fontWeight: 'bold',
    color: 'maroon',
  },
});