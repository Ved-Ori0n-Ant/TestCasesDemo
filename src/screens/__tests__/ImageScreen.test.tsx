import React from "react";
import { renderWithNavigation } from "../../utils/renderWithNavigation";
import ImageScreen from "../ImageScreen";

it('should render', () => {
    const tree = renderWithNavigation(<ImageScreen />)
    expect(tree.toJSON()).toMatchSnapshot();
});

it('should have image view', () => {
    const tree = renderWithNavigation(<ImageScreen />)
    const image = tree.getByTestId('Image')
    expect(image).not.toBeNull();
});

it('should have image url', () => {
    const tree = renderWithNavigation(<ImageScreen />)
    const image = tree.getByTestId('Image')
    console.log('---', image.props.source.uri)
    expect(image.props.source.uri).toBe('https://reactnative.dev/docs/assets/p_cat2.png');
})