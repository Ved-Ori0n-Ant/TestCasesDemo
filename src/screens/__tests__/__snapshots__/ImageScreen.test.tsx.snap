// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://reactnative.dev/docs/assets/p_cat2.png",
        }
      }
      style={
        {
          "height": 200,
          "width": 200,
        }
      }
      testID="Image"
    />
  </View>
  <View
    style={
      {
        "backgroundColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
    testID="np_btn"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Previous_btn"
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Previous
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Next_btn"
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
